commons/src/main/java/io/aiven/kafka/connect/common/config/CompressionFragment.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.config;
+
+import java.util.Objects;
+
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef;
+
+import io.aiven.kafka.connect.common.config.validators.FileCompressionTypeValidator;
+
+/**
+ * The configuration fragment that defines the compression characteristics.
```

Comment:
I think this class is for all common configs between sink and source, and not specific to compression ?

Reply:
This class implements the Compression Fragment so it can be re-useable in multiple places, it can be used in both the sink and source, but also it is used in the FileNameFragment as well to determine file extension names, so this allows us to have a single place where we maintain the code!